game-core/src/main/java/games/strategy/triplea/ai/pro/ProPurchaseAi.java

```diff
@@ -2339,6 +2338,13 @@ private void upgradeUnitsWithRemainingPUs(
     }
   }
 
+  private static double findUpgradeUnitEfficiency(
```

Comment:
Any objection to javadoc'ing this method? I'm thinking capturing the context of what we are doing, the motivation/philosophy behind this calculation would be useful context to maintain going forward? 

For example, it's a bit curious why/when we can swap defense efficiency for movement

Reply:
Added.